File: src/hashmaps-and-sets/pair-sum-unsorted.ts
export function pairSumUnsorted(nums: number[], target: number): number[] {
    const numToIndex = new Map<number, number>()
    for (let i = 0; i < nums.length; i++) {
        const complementNum = target - nums[i]
        if (numToIndex.has(complementNum)) {
            return [numToIndex.get(complementNum)!, i]
        }
        numToIndex.set(nums[i], i)
    }

    return []
}

File: src/hashmaps-and-sets/pair_sum_unsorted.go
package hashmapsandsets

func PairSumUnsorted(nums []int, target int) []int {
	numToIndex := make(map[int]int)
	for i := 0; i < len(nums); i++ {
		complementNum := target - nums[i]
		if complementIndex, ok := numToIndex[complementNum]; ok {
			return []int{complementIndex, i}
		}
		numToIndex[nums[i]] = i
	}

	return []int{}
}
